src/bank.rs

```diff
@@ -306,6 +306,41 @@ impl Bank {
         res
     }
 
+    /// Process an ordered list of entries.
+    pub fn process_entries_tail(
+        &self,
+        entries: Vec<Entry>,
+        tail: &mut Vec<Entry>,
+        tail_idx: &mut usize,
+    ) -> Result<u64> {
+        let mut entry_count = 0;
+
+        for entry in entries {
+            entry_count += 1;
+
+            if tail.len() > *tail_idx {
+                tail[*tail_idx] = entry.clone();
+            } else {
+                tail.push(entry.clone());
+            }
+            *tail_idx = (*tail_idx + 1) % WINDOW_SIZE as usize;
+
+            if !entry.transactions.is_empty() {
+                for result in self.process_transactions(entry.transactions) {
+                    result?;
+                }
+            }
+            // TODO: verify this is ok in cases like:
```

Comment:
Is this comment relevant anymore? In no, can you delete?

Reply:
I think the comment is still relevant, because I don't know the answer

Reply:
The answer is that we don't need to register any entry IDs for security purposes. We register them for only these reasons:

* So that the server doesn't need to compare against every signature there's ever been to ensure a transaction is not a duplicate. It can look up a set of signatures by `last_id`.
* So that a client knows its transaction is only valid on a ledger containing its transaction's `last_id` or later.
* So that a client knows that if a transaction wasn't accepted after a certain amount of time that it'll never be accepted.